qiskit/visualization/matplotlib.py

```diff
@@ -285,8 +290,6 @@ def param_parse(self, v):
                 param_parts[i] = '$-$' + param_parts[i][1:]
 
         param_parts = ', '.join(param_parts)
-        # remove $'s since "${}$".format will add them back on the outside
-        param_parts = param_parts.replace('$', '')
         param_parts = param_parts.replace('-', u'\u02d7')
```

Comment:
There was an issue with hyphens and minus sign widths when everything was mathtext, so line 290 can be removed and minus signs will go back to being wider.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/4669/commits/0998cd71f936fb12188ee92dc545952cd386c2ad